osgar/node.py

```diff
@@ -0,0 +1,46 @@
+"""
+  Node - processing unit
+"""
+from threading import Thread
+
+from osgar.bus import BusShutdownException
+
+
+class Node(Thread):
+    """
+       Parent classs for all processing "nodes".
+    """
+    def __init__(self, config, bus):
+        super().__init__()
+        self.setDaemon(True)
+        self.bus = bus
+        self.time = None
+
+    def publish(self, channel, data):
+        return self.bus.publish(channel, data)
+
+    def listen(self):
+        return self.bus.listen()
+
+    def sleep(self, secs):
+        self.bus.sleep(secs)
+
+    def is_alive(self):
+        return self.bus._is_alive
+
+    def update(self):
+        timestamp, channel, data = self.bus.listen()
+        self.time = timestamp
+        setattr(Node, channel, data)
```

Comment:
This is not how `setattr` works, please see https://docs.python.org/3.5/library/functions.html#setattr.
```suggestion
        setattr(self, channel, data)
```

Reply:
Thanks, fixed with unittest. There is maybe one more detail, that the variables are not existing until the first `update()` of that particular "channel" ... so maybe we should initialize them in `__init__()` from "bus" input configuration ... but what to do, if you do not want them??

Reply:
I don't know. Let's keep it the way it is for now. When we refactor more code, we have a better understanding of what is needed.